File: backend/dotnet/Models/ProjectMember.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace dotnet.Models
{
    [Table("project_members")]
    public class ProjectMember
    {
        [Key]
        [Required]
        [Column("project_member_id")]
        public string Id { get; set; } = string.Empty;

        [Required]
        [ForeignKey("Project")]
        [Column("project_id")]
        public string ProjectId { get; set; } = string.Empty;
        public Project Project { get; set; }

        [Required]
        [ForeignKey("User")]
        [Column("email")]
        public string Email { get; set; } = string.Empty;
        public User User { get; set; }
    }
}

File: backend/springboot/src/main/java/tn/maiko26/springboot/models/relations/ProjectMember.java
package tn.maiko26.springboot.models.relations;

import jakarta.persistence.*;
import tn.maiko26.springboot.models.Project;
import tn.maiko26.springboot.models.User;

import java.io.Serializable;

@Entity
@Table(name = "project_members")
@IdClass(ProjectMemberId.class)
public class ProjectMember {
    @Id
    @ManyToOne
    @JoinColumn(name = "project_id")
    private Project project;

    @Id
    @ManyToOne
    @JoinColumn(name = "email", referencedColumnName = "email")
    private User user;
}

class ProjectMemberId implements Serializable {
    private String project;
    private String user;
}